static/src/javascripts/projects/common/modules/identity/email-sign-in-banner/index.js

```diff
@@ -0,0 +1,87 @@
+// @flow
+
+import { Message, hasUserAcknowledgedBanner } from 'common/modules/ui/message';
+import { trackNonClickInteraction } from 'common/modules/analytics/google';
+import ophan from 'ophan/ng';
+import config from 'lib/config';
+import userPrefs from 'common/modules/user-prefs';
+import type { Banner } from 'common/modules/ui/bannerPicker';
+import { campaigns } from './campaigns';
+import { make as makeTemplate, messageCode } from './template';
+import type { Campaign } from './campaigns';
+
+const displayEventKey: string = `${messageCode} : display`;
+const userPrefsReferrerKey = 'emailbanner.referrerEmail';
+
+const emailPrefsLink = `https://${config.get('page.host')}/email-newsletters`;
+
+const signInLink = `${config.get('page.idUrl')}/login?returnUrl=${config.get(
+    emailPrefsLink
+)}`;
+
+const getEmailCampaignFromUtm = (utm): ?Campaign =>
+    campaigns.find((campaign: Campaign) => campaign.utm === utm);
+
+const getEmailCampaignFromUrl = (): ?Campaign => {
+    const emailCampaignInUrl = (new window.URLSearchParams(
+        window.location.search
+    ).getAll('utm_campaign') || [''])[0];
+    return getEmailCampaignFromUtm(emailCampaignInUrl);
+};
+
+const isSecondEmailPageview = (): boolean =>
+    userPrefs.get(userPrefsReferrerKey) !== null;
+
+const isInExperiment = (): boolean =>
+    config.get('switches.idEmailSignInUpsell', false);
```

Comment:
I don’t remember, but kinda surprises me that there’s no common functionality for “is this switch on” apart from looking straight in config - even if that’s quite a simple look up.

Reply:
there's one for `userPrefs` switches but not one for the global config AFAIK